src/components/ChooseOpponentDropdown.tsx

```diff
@@ -0,0 +1,64 @@
+import { useRef, useState } from "react";
+
+import { type IPlayer } from "../api/player/get/types";
+import useClickOutside from "../hooks/useClickOutside";
+import { useOpponentStore, useUserStore } from "../store";
+
+export const ChooseOpponentsDropdown = ({
+  playersList,
+  userName,
+}: {
+  playersList?: IPlayer[];
```

Comment:
```suggestion
  playersList: IPlayer[] | undefined;
```

because, likely, this is not an optional prop of the `ChooseOpponentsDropdown`

Reply:
+